### PR TITLE
Add conditional colour assignment to date tile icon.

### PR DESCRIPTION
**Pull request by HeinzUBC**
If there is at least 1 TODOItem that is not finished for a particular date tile, that icon will be coloured red. If all the TODOItems are finished for a particular date tile, that icon will be coloured grey.

Grey icon use case,

Click grey icon 1

Click grey icon 2

Red icon use case,

Click red icon 1

Click red icon 2

### CODE REVIEW

File: client/src/components/TODOList/TODOCalendarIcon.js [hunk 1]
```diff
@@ -11,10 +11,26 @@ import TODOItem from "./TODOItem";
...
+  const configureTODOIconColour = () => {
+    const notFinishedTODOs = TODOList.filter((todo) => {
+      return todo.isFinished === false;
+    });
+
+    if (notFinishedTODOs.length > 0) {
+      return "fill-red-500";
+    }
+
+    return "fill-gray-400";
```
Comment (leungkenneth) [inline client/src/components/TODOList/TODOCalendarIcon.js:27]: The new color assignment functionality seems to be working as intended so I would approve off of that. However, I did some inspect element toggling and seemingly managed to get the number disappearing due to styling fixed. Perhaps you can adapt the changes I have made here in your css file:

image

That transparency on hover makes the number disappear on hover, the transparency on the calendar tile makes the whole tile transparent, and setting the background-color and color to transparent when the tile is active makes the number disappear when the date is selected in general.

Comment (HeinzUBC) [reply client/src/components/TODOList/TODOCalendarIcon.js:27]: wait... why is do I see blue in your image? I tried to get rid of that earlier

**Comment by HeinzUBC** [comment]
Here is the fixed behaviour of the Calendar 😀

Screen Shot 2023-07-28 at 10 23 55 PM

Screen Shot 2023-07-28 at 10 24 14 PM

File: client/src/components/TODOList/TODOCalendarIcon.js (hunk 1 above)
Comment (HeinzUBC) [reply client/src/components/TODOList/TODOCalendarIcon.js:27]: I have fixed it all haha

**Review by awctw — APPROVED** [review]

**Pull request merged**